crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -247,7 +244,7 @@ impl<'src> Parser<'src> {
     /// See <https://matklad.github.io/2020/04/13/simple-but-powerful-pratt-parsing.html>
     fn parse_expression_with_precedence(&mut self, previous_precedence: Precedence) -> ParsedExpr {
```

Comment:
I think the comment here is outdated because it now parses expressions of arbitrary precedence. 

I also find the method name somewhat confusing because this method might parse multiple expressions in case it is a binary expression. I wonder if we should rename the method to `parse_binary_expression_or_higher`

Reply:
It does _still_ uses the Pratt Parsing algorithm if that's what you mean by the comment.

I think the only confusing part is about LHS parsing which I found confusing as well. I referred to `rust-analyzer` and I think it does the same thing ([source](https://github.com/rust-lang/rust-analyzer/blob/0140272daac91322b090a3cc171861fb8e49c427/crates/parser/src/grammar/expressions.rs#L329-L403)).

Reply:
I'll think about this, and probably update the documentation, but for now I'll merge this as is.